lib/blocklist.c

```diff
@@ -102,8 +104,33 @@ void allowlist_prefix(char *ip, int prefix_len)
 	_add_constraint(addr, prefix_len, ADDR_ALLOWED);
 }
 
+static int is_ip_ipv6(char *ip) {
+	// don't modify the input string
+	int length = strlen(ip);
+	char *new_str = (char*)malloc(length);
+	if (new_str == NULL) {
+		log_fatal("blocklist", "could not allocate memory for IPv6 validation");
+	}
+	strncpy(new_str, ip, length);
+	// check if there's a subnet maskChar
+	char *maskChar = strchr(new_str, '/');
+	if (maskChar != NULL) {
+		// set maskChar char to NULL char so we can check if subnet is valid IPv6
+		*maskChar = '\0';
+	}
+	// attempt conversion of IP into IPv6 struct to check if IP is an IPv6 address
+	struct in6_addr ipv6_addr;
+	if (inet_pton(AF_INET6, new_str, &ipv6_addr) == 1) {
+		return true;
+	}
+	return false;
+}
+
 static int init_from_string(char *ip, int value)
 {
+	if (is_ip_ipv6(ip)) {
+		log_fatal("constraint", "%s IPv6 subnets are not supported", ip);
```

Comment:
I think that we want to ignore IPv6 subnets, not fatal on them. The idea is that many blacklists will have both IPv4 and IPv6 entries in them. Since ZMap is an IPv4-only tool right now, we simply want to ignore the IPv6 subnets.